data_safe_haven/commands/config.py

```diff
@@ -51,6 +53,51 @@ def show_shm(
 
 
 # Commands related to an SRE
+@config_command_group.command()
+def available() -> None:
+    """List the available SRE configurations for the selected Data Safe Haven context"""
+    logger = get_logger()
+    try:
+        context = ContextManager.from_file().assert_context()
+    except DataSafeHavenConfigError as exc:
+        logger.critical(
+            "No context is selected. Use `dsh context add` to create a context "
+            "or `dsh context switch` to select one."
+        )
+        raise typer.Exit(1) from exc
+    azure_sdk = AzureSdk(context.subscription_name)
+    try:
+        blobs = azure_sdk.list_blobs(
+            container_name=context.storage_container_name,
+            prefix="sre",
+            resource_group_name=context.resource_group_name,
+            storage_account_name=context.storage_account_name,
+        )
+    except DataSafeHavenAzureStorageError as exc:
+        logger.critical("Ensure SHM is deployed before attempting to use SRE configs.")
+        raise typer.Exit(1) from exc
+    if not blobs:
+        logger.critical(f"No configurations found for context '{context.name}'.")
+        raise typer.Exit(1)
+    pulumi_config = DSHPulumiConfig.from_remote(context)
+    console.print(f"Available SRE configurations for context '{context.name}':")
+    for blob in blobs:
+        sre_config = SREConfig.from_remote_by_name(
+            context, blob.removeprefix("sre-").removesuffix(".yaml")
+        )
+        stack = SREProjectManager(
+            context=context,
+            config=sre_config,
+            pulumi_config=pulumi_config,
+            create_project=True,
+        )
+        stack_outputs = stack.run_pulumi_command("stack output")
+        if "No output values" in stack_outputs:
+            console.print(f"SRE {sre_config.name} not deployed")
+        else:
+            console.print(f"SRE {sre_config.name} deployed")
```

Comment:
If you want nicer output here, you could instead store a dictionary of `sre_name` -> `is_deployed?` and use the logic from `UserHander::list()` to print this as a table. Something like:

```python
headers = ["SRE name", "Deployed?"]
rows = [[name, "x" if is_deployed else ""] for name, is_deployed in my_dict.items()]
console.tabulate(headers, rows)
```

Reply:
Was considering something like that!

Reply:
updated output above!